src/capi_frontend/capi.cpp

```diff
@@ -172,6 +175,79 @@ void OVMS_ServerDelete(OVMS_Server* server) {
     // delete passed in ptr once multi server configuration is done
 }
 
+OVMS_Status* OVMS_GetMetadataByPointer(OVMS_Metadata* metadata, const char* pointer, const char** value, size_t* size) {
+    if (metadata == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "metadata"));
+    }
+    if (pointer == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "pointer"));
+    }
+    if (value == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "base"));
+    }
+    if (size == nullptr) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "byte_size"));
+    }
+    rapidjson::Document* doc = reinterpret_cast<rapidjson::Document*>(metadata);
+    rapidjson::Value* val = rapidjson::Pointer(pointer).Get(*doc);
+    if (!val) {
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_PTR, "value not found"));
```

Comment:
NONEXISTENT_PTR is reserved for scenarios when users pass nullptr, here this is rapidjson returning nullptr